theme/admin/builder.php

```diff
@@ -0,0 +1,147 @@
+
+		function mango_deploy_getDataset() {
+			return [
+				'templates' => [
+					'src/templates/*.jade',
+				],
+				'data' => [
+		 			'site' => mango_expandPaths(get_option('site_settings')),
+				],
+			];
+		}
+
+	2. Fill in configuration paremeters:builder options
+
+	3. Test it and profit
+
+ */
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+
+function mango_escape($str)
+{
+	return htmlspecialchars($str, ENT_COMPAT | ENT_HTML5);
+}
+
+function mango_pipeTransformer($key, $value, $transformers) {
+	if(!empty($transformers[$key])) {
+		return $transformers[$key]($value);
+	}
+	return $value;
+}
+
+function mango_expandPaths($root, $delimiter = NULL, $transformers = []) {
+	if(!$delimiter) {
+		$delimiter = '__';
+	}
+	if(!is_array($root)) {
+		return $root;
+	}
+	$updated = [];
+	foreach($root as $key => $val) {
+		$keyParts = explode($delimiter, (string) $key);
+		if(is_string($key) && count($keyParts) > 1) {
+			$firstKeyPart = array_shift($keyParts);
+			$keyRest = implode($delimiter, $keyParts);
+			$updated[$firstKeyPart] = isset($updated[$firstKeyPart]) ? $updated[$firstKeyPart] : [];
+			// to-do: optimize
+			$updated[$firstKeyPart][$keyRest] = mango_pipeTransformer($keyRest, $val, $transformers);
+			$updated[$firstKeyPart] = mango_pipeTransformer($key, mango_expandPaths($updated[$firstKeyPart], $delimiter, $transformers), $transformers);
+		}
+		else {
+			$updated[$key] = mango_pipeTransformer($key, mango_expandPaths($val, $delimiter, $transformers), $transformers);
+		}
+	}
+	return $updated;
+}
+
+
+function mango_deployPage() {
+	global $App;
+	$config = $App->parameters['builder'];
+
+	view('../admin/builder', [
+		"revision" => $config['revision'],
+		"subfolder" => $config['aws']['subfolder'],
+		"preview" => json_encode(mango_deploy_getDataset(), JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT),
+	]);
+}
+
+function mango_triggerDeploy() {
+	global $App;
+	$config = $App->parameters['builder'];
+	$httpRequest = $App->getByType('Nette\Http\Request');
+
+	$dataset = mango_deploy_getDataset();
+
+	echo "<h1>Starting deploy</h1>";
+	dump($httpRequest->getPost());
+	echo "<p>It might take a while...</p>";
+
+	ob_end_flush();
+	flush();
+
+	$payload = [
+		'remote' => $config['remote'],
+		'revision' => $httpRequest->getPost('revision'),
+		'mango-cli' => $config['mango-cli'],
+		'aws' => [
+			'bucket' => $config['aws']['bucket'],
+			'accessKeyID' => $config['aws']['key'],
+			'secretAccessKey' => $config['aws']['secret'],
+			'subfolder' => $httpRequest->getPost('subfolder'),
+		],
+		'dataset' => $dataset,
+	];
+
+	$client = new Client([ 'base_uri' => $config['url'], 'timeout'  => 0	]);
+
+	try {
+		$response = $client->post('/upload', ['json' => $payload]);
+	} catch (BadResponseException $e) {
+		$response = $e->getResponse();
+	}
+
+	$responseBody = json_decode((string) $response->getBody());
+
+	if ($responseBody->status != 'success') {
```

Comment:
Právě proto, že chceš string, tak je `!==` vhodnější. Pokud v `$responseBody->status` nebude string, tak to selže, vrátí `false`, v obou případech (`!=` i `===`). S `!==` dáváš ostatním programátorům vědět, že to má selhat i kdyby na pravé straně byl prázdný řetězec a že neočekáváš žádnou magii se změnou typů. Už jsme to řešil v javascriptu.